grammars/repositories/flavors/rmarkdown.cson

```diff
@@ -5,27 +5,32 @@
 key: 'rmarkdown'
 patterns: [
   {
+    # test and edit regex on https://regex101.com/r/PDu8pO/1/tests
     begin: '(?x)
-      (```)(
-        (\\{)(r)(?:\\s)?
-        (?:([a-zA-Z0-9|_|-]*)(?=[\\s|,|}])\\s?)?
-        ([^\\}]*)?
-        (\\})
-      )
+        (?:\\n|^)(`{3,})
+            (\\{)
+                (r|javascript|js|jsx|python|sql|bash|rcpp|stan|css)[ ]*,?[ ]*
```

Comment:
My command-line git is horrible, so I haven't been able to test these changes. I think however that the `[ ]*,?[ ]*` bit (also on the next line) causes the issues with the three tests that fail.

I haven't tested any of the R Markdown syntax in R Studio. Have you? I'm curious if the extra white-spaces, also when having a key=value pair in the parameters, is an issue.